src/network/networkd-dhcp-common.c

```diff
@@ -469,6 +469,85 @@ int config_parse_dhcp_send_option(
         return 0;
 }
 
+int config_parse_dhcp_request_options(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        Network *network = data;
+        const char *p;
+        int r;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+
+        if (isempty(rvalue)) {
+                if (streq(section, "DHCPv4"))
+                        network->dhcp_request_options = set_free(network->dhcp_request_options);
+                else
+                        network->dhcp6_request_options = set_free(network->dhcp6_request_options);
```

Comment:
i think it would be nicer to use the `ltype` argument for distuingishing DHCPv4 and DHCPv6 here. I.e. set the fourth column in the gperf files to AF_INET or AF_INET6, to chose between dhcpv4 and dhcp6. The fourth column can be any arbitrary integer, and passing AF_INET and AF_INET6 kinda makes sense here. The column is passed as it is as ltype to this handler function.

This doesn't change much of course, the section is as good for comparing, but in theory at least comparing an integer is still a bit quicker than comparing a string...